segbot_arm_tasks/src/object_to_office_task.cpp

```diff
@@ -0,0 +1,341 @@
+		exit(1);
+	}
+		
+	//select the object with most points as the target object
+	int largest_pc_index = -1;
+	int largest_num_points = -1;
+	for (unsigned int i = 0; i < table_scene.cloud_clusters.size(); i++){
+			
+		int num_points_i = table_scene.cloud_clusters[i].height* table_scene.cloud_clusters[i].width;
+		
+		if (num_points_i > largest_num_points){
+			largest_num_points = num_points_i;
+			largest_pc_index = i;
+		}
+	}
+		
+	//Step 8: call the grasp action
+	
+	//create the action client
+	actionlib::SimpleActionClient<segbot_arm_manipulation::TabletopGraspAction> ac_grasp("segbot_tabletop_grasp_as",true);
+	ac_grasp.waitForServer();
+		
+	pressEnter("Press [ENTER] to proceed");
+
+	
+	//create and fill goal
+	segbot_arm_manipulation::TabletopGraspGoal grasp_goal;
+	grasp_goal.cloud_plane = table_scene.cloud_plane;
+	grasp_goal.cloud_plane_coef = table_scene.cloud_plane_coef;
+	for (unsigned int i = 0; i < table_scene.cloud_clusters.size(); i++){
+		grasp_goal.cloud_clusters.push_back(table_scene.cloud_clusters[i]);
+	}
+	grasp_goal.target_object_cluster_index = largest_pc_index;
+	grasp_goal.action_name = segbot_arm_manipulation::TabletopGraspGoal::GRASP;
+	grasp_goal.grasp_selection_method=segbot_arm_manipulation::TabletopGraspGoal::CLOSEST_ORIENTATION_SELECTION;
+			
+	//send the goal
+	ROS_INFO("Sending goal to action server...");
+	ac_grasp.sendGoal(grasp_goal);
+	ac_grasp.waitForResult();
+	ROS_INFO("Action Finished...");
+
+	pressEnter("Press [ENTER] to proceed");
+
+
+	//next, lift and make arm safe again
+	lift(n,0.05);
+	
+	pressEnter("Press [ENTER] to proceed");
+
+	
+	safe = segbot_arm_manipulation::makeSafeForTravel(n);
+	if (!safe)
+		return 1;
+		
+	pressEnter("Press [ENTER] to proceed");
+
+	
+	//next, back out
+	segbot_arm_manipulation::TabletopApproachGoal back_out_goal;
+	back_out_goal.command = "back_out";
+	ac_approach.sendGoal(back_out_goal);
+	ac_approach.waitForResult();
+	
+	//next go to an office - TO DO
+	std::string location = "d3_414a1";
+
+	bwi_kr_execution::ExecutePlanGoal goal_asp;
+    bwi_kr_execution::AspRule rule;
+    bwi_kr_execution::AspFluent fluent;
+    fluent.name = "not facing";
+    fluent.variables.push_back(location);
+	
+	actionlib::SimpleActionClient<bwi_kr_execution::ExecutePlanAction> client_asp("/action_executor/execute_plan", true);
+	client_asp.waitForServer();
+
+	rule.body.push_back(fluent);
+    goal_asp.aspGoal.push_back(rule);
+
+    ROS_INFO("sending goal");
+    client_asp.sendGoalAndWait(goal_asp);
+	
+	//hand over object
+	pressEnter("Press [ENTER] to proceed");
+
+	
+	segbot_arm_manipulation::moveToPoseMoveIt(n,pose_outofview);
+	
+	segbot_arm_manipulation::TabletopGraspGoal handover_goal;
+	handover_goal.action_name = segbot_arm_manipulation::TabletopGraspGoal::HANDOVER;
+	handover_goal.timeout_seconds = 30.0;
+	
+	ac_grasp.sendGoal(handover_goal);
+	ac_grasp.waitForResult();
+	
+	
+	
+		//lift and lower the object a bit, let it go and move back
+		/*lift(n,0.07);
```

Comment:
if I leave something commented it's because I may need to uncomment it during testing